contracts/common/SecuredTokenTransfer.sol

```diff
@@ -18,21 +18,23 @@ abstract contract SecuredTokenTransfer {
     function transferToken(address token, address receiver, uint256 amount) internal returns (bool transferred) {
         // 0xa9059cbb - keccack("transfer(address,uint256)")
         bytes memory data = abi.encodeWithSelector(0xa9059cbb, receiver, amount);
-        // solhint-disable-next-line no-inline-assembly
+        /* solhint-disable no-inline-assembly */
+        /// @solidity memory-safe-assembly
         assembly {
-            // We write the return value to scratch space.
```

Comment:
scratch space is memory safe, see: https://docs.soliditylang.org/en/latest/internals/layout_in_memory.html

What is key here is that we only ever write 32 bytes (which is smaller than the 64-byte scratch space).

Reply:
Specifically, from https://docs.soliditylang.org/en/latest/assembly.html#memory-safety:

> In particular, a memory-safe assembly block may only access the following memory ranges:
> * [...]
> * The scratch space between memory offset 0 and 64 mentioned above.

Reply:
Will revert! 👍🏾